### DISCUSSION TITLE
Creating the $APPDATA directory?

### DISCUSSION THREAD

**Discussion by aleph1** [tauri-apps/tauri#6750] [Q&A]
I am able to read and write files using "fs" and the "scope" array in tauri.conf.json, however, I still can’t determine how to create a directory at $APP/{identifier} where {identifier} is the bundle identifier, without giving "fs" full read/write access to the $APP directory. Since Tauri suggests giving as little access as possible using the API and scopes, it seems counterintuitive to give the application access to all of $APP just to be able to create a directory with the bundle identifier.

Is there a better way to do this, without having to add the scope $APP?

**Comment by FabianLars** [comment]
`$APP/{identifier}` would result in a path that includes the bundle-identifier twice, since $APP is just an alias for $APPCONFIG (similar to $APPDATA with just Linux having different paths).

Anyway, to answer your question, to create the directory, first add `$APPDATA/` to the scope. Note that this will only add the folder to the scope, not its content!
Then you can do something like this:
```js
await createDir(await appDataDir());
```

**Reply by eyyyyyyy3** [reply]
And how would I go on about managing the content, meaning if I have some files like language files or idk images or a setttings file (you get the idea) how would I add those to my project and ensure that when I build my project and have a installer the user, when installing the app with the installer,  would have the same files at the same directory. Thx

**Reply by FabianLars** [reply]
You can embed additional with tauri's resource feature https://tauri.app/v1/guides/building/resources

For read-only files this would already be enough. FIles you need write access to need to be copied over first, but you'll have to do that on first app start, or whenever you try to access them (check if they're there, if not get them from the root).
That said, on Windows you could modify the installer to copy the files in there for you but as soon as you want macos or linux support this won't work anymore (same for a potential portable windows mode).

**Reply by eyyyyyyy3** [reply]
If I want to have write access, where exactly in my project should I store those files initially? In the src directory or in src-tauri and should I make additional folders within those directories?

**Reply by FabianLars** [reply]
doesn't matter, it's basically just personal preference. You can include them from wherever you want using the resource feature and it has no influence on whether they're writable or not either.

**Reply by eyyyyyyy3** [reply]
is there a standard or a recommendation from your team?

**Reply by FabianLars** [reply]
Not really as long as it's not inside a `src/` dir. I think we typically use a folder in src-tauri like `src-tauri/assets/` or whatever, and community members with a js heavy background often go for a similar folder _next_ to src-tauri.
